app/vlagent/kubernetescollector/collector.go

```diff
@@ -23,6 +23,11 @@ type kubernetesCollector struct {
 
 	currentNode node
 
+	// namespaces caches metadata for all namespaces in the cluster.
+	// The map is refreshed if a pod references an unknown namespace.
+	// No lock is needed since only the event handler goroutine accesses this map.
```

Comment:
This comment is no longer correct after the commit ed9e1c071d035eba6ebcbb8580ee09f005b2ea69 , where the `kubernetesCollector` fields can be updated and read from concurrently running goroutines.

@vadimalekseev , please address this in separate commits if there are data races after the commit ed9e1c071d035eba6ebcbb8580ee09f005b2ea69 .

Reply:
done